src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -809,19 +809,23 @@ ModifiesLocalTableWithRemoteCitusLocalTable(List *rangeTableList)
 
 
 /*
- * IsTableLocallyAccessible returns true if the given table
- * can be accessed in local.
+ * IsLocallyAccessibleCitusLocalTable returns true if the given table
+ * is a citus local table that can be accessed using local execution.
  */
 static bool
-IsTableLocallyAccessible(Oid relationId)
+IsLocallyAccessibleCitusLocalTable(Oid relationId)
 {
-	if (!IsCitusTable(relationId))
+	if (IsCitusTableType(relationId, CITUS_LOCAL_TABLE))
 	{
-		/* local tables are locally accessible */
-		return true;
+		return false;
```

Comment:
why do we return false if it is a citus local table?

Reply:
It missed a `!`.